### ISSUE TITLE
AutoMapperConfig::getClassDistance should consider interfaces

### ISSUE THREAD

**Issue by fnash** [mark-gerarts/automapper-plus#35]
Hello,

It would be nice registering mapping for an interface, making the code below possible for unknown types:

```php
$config->registerMapping(ObjectInterface::class, MyDTO::class);


$dtos = [];
foreach ($objects as $object) {
    $dtos[] = $automapper->map($object, MyDTO::class);
}

```

AutoMapperConfig::getClassDistance should consider interfaces with **class_implements** if no result with **class_parents**

**Comment by mark-gerarts** [comment]
Well, I hadn't actually considered this before, but it sounds like a must-have feature. The important thing to keep in mind is (as you suggested) that we only check for the interface if we don't find a registered class. This because a class is more specific than an interface.

I'll take a look into this in the coming weeks, I'll keep you posted. Thanks for the suggestion!

**Comment by BoShurik** [comment]
Note, that interface is about methods, so we don't know about properties in such case. Looks like we should allow only `MapFrom` operation

**Comment by fnash** [comment]
@BoShurik I didn't read the code. Do automapper actually know about properties if a mapping is registered for a concrete class?
If it is done at runtime for classes so it must be the same for interfaces.
I understand your point of view but I think it is up to the developer to assume errors of a mapping for an interface.
Anyway, interface may not have methods. It is only about contract. And that is my use case.
What do you think?

**Comment by mark-gerarts** [comment]
You're right @BoShurik, since interfaces only specify which methods are available we don't know anything about the properties. Next to `MapFrom`, `Ignore` would be a perfectly valid operation as well (there might be others). 

But I can totally understand @fnash's opinion as well. Leave the user responsible to provide sensible input to the mapper, not restricting their ability to define operations.

First things first I'll work on implementing interfaces. I'll decide on whether or not we'll restrict the operations afterwards.

**Comment by mark-gerarts** [comment]
Using interfaces is available in 1.3.5 now. There are no restrictions on the operations, but  this might be added in the future.
